apps/src/lib/client/utils.rs

```diff
@@ -324,26 +324,23 @@ pub async fn join_network(
         tokio::task::spawn_blocking(move || {
             let mut config = Config::load(&base_dir, &chain_id, None);
 
-            config.ledger.tendermint.tendermint_mode =
-                TendermintMode::Validator;
+            config.ledger.shell.tendermint_mode = TendermintMode::Validator;
             // Validator node should turned off peer exchange reactor
-            config.ledger.tendermint.p2p_pex = false;
-            // Remove self from persistent peers
-            config
-                .ledger
-                .tendermint
-                .p2p_persistent_peers
-                .retain(|peer| {
-                    if let TendermintAddress::Tcp {
-                        peer_id: Some(peer_id),
-                        ..
-                    } = peer
-                    {
-                        node_id != *peer_id
-                    } else {
-                        true
-                    }
-                });
+            // fixme: we shouldn't be setting config values that are settable in
+            // the actual config.toml config.ledger.
+            // tendermint_config.p2p.pex = false; Remove self from
+            // persistent peers
```

Comment:
I think pex should still be switched off for validators by default - any reason to comment it out here?

Reply:
cause if we force it here, we can't modify it via config